include/ffi.h.in

```diff
@@ -357,6 +357,11 @@ ffi_prep_closure_loc (ffi_closure*,
 		      void *user_data,
 		      void*codeloc);
 
+#if defined(__x86_64__) || defined(__arm64__)
```

Comment:
Would it make sense to make this available on all architectures?

Reply:
If this seems valuable outside of our internal use case, then I certainly can.  One of the main reasons I'm putting all this up is to see if any of this work is valuable to the larger OSS community or should remain Apple-internal changes for our specific cases.

We only wrote the implementation in src/aarch64/ffi.c and arc/x86/ffi64.c since that is all we use, but it should be usable elsewhere.  This was necessitated by the trampoline changes, and I think we ended up using it in python3 and a few other places where there were assumptions that codeloc == closure.

Reply:
Ahh I see. That's a good question for the maintainers. /cc @atgreen @tromey

Reply:
Leave it for now.  We may come back to this as we need to do something for the Linux architectures.

Reply:
Ok.  I'll break that out into a separate PR that adds the API to all archs.

@atgreen What are your opinions on the API naming?  Is ffi_find_closure_for_code() ok, or would you prefer something different?